goal_src/levels/maincave/mother-spider.gc

```diff
@@ -5,3 +5,2134 @@
+        (joint-control-channel-group-eval! a0-2 (the-as art-joint-anim #f) num-func-seek!)
+        )
+      )
+    (let ((v1-22 (-> self mode)))
+      (cond
+        ((zero? v1-22)
+         (go mother-spider-idle)
+         )
+        ((= v1-22 1)
+         (go mother-spider-tracking)
+         )
+        ((= v1-22 2)
+         (set! (-> self started-birthing-time) (the-as uint (-> *display* base-frame-counter)))
+         (set! (-> self birthing-counter) (max 0 (- (-> self max-baby-count) (-> self baby-count))))
+         (go mother-spider-birthing)
+         )
+        )
+      )
+    (none)
+    )
+  :post
+  (the-as (function none :behavior mother-spider) transform-post)
+  )
+
+(defstate mother-spider-tracking (mother-spider)
+  :event
+  mother-spider-default-event-handler
+  :enter
+  (behavior ()
+    (set! (-> self hit?) #f)
+    (set! (-> self state-time) (-> *display* base-frame-counter))
+    (set! (-> self last-player-in-air-time) (the-as uint (-> *display* base-frame-counter)))
+    (none)
+    )
+  :trans
+  (behavior ()
+    (if (-> self hit?)
+        (go mother-spider-hit-while-tracking)
+        )
+    (if (letgo-player? self)
+        (go mother-spider-traveling (the-as uint 0))
+        )
+    (let ((gp-0 (new 'stack-no-clear 'vector)))
+      (vector<-cspace! gp-0 (-> self node-list data 6))
+      (if (or (logtest? (-> *target* state-flags) #x80f8) (< (vector-vector-distance gp-0 (camera-pos)) 8192.0))
+          (go mother-spider-traveling (the-as uint 2))
+          )
+      )
+    (when (is-player-stuck? self)
+      (let ((f1-4 (fmax
+                    (fmin (+ -28672.0 (-> self player-sticky-dist-from-anchor)) (-> self max-dist-from-anchor))
+                    (-> self idle-dist-from-anchor)
+                    )
+                  )
+            )
+        (if (< 409.6 (fabs (- f1-4 (-> self dist-from-anchor))))
+            (go mother-spider-traveling (the-as uint 1))
+            )
+        )
+      )
+    (if (or (not *target*) (and
+                               (logtest? (-> *target* control unknown-surface00 flags) 2048)
+                               (zero? (logand (-> *target* control status) 1))
+                               )
+              )
+        (set! (-> self last-player-in-air-time) (the-as uint (-> *display* base-frame-counter)))
+        )
+    (when (and
+            *target*
+            (>= (- (-> *display* base-frame-counter) (the-as int (-> self last-spit-time))) 900)
+            (>= (- (-> *display* base-frame-counter) (the-as int (-> self last-player-in-air-time))) 18)
+            (>= (-> self max-spit-xz-dist) (vector-vector-xz-distance (-> self root-override trans) (target-pos 0)))
+            )
+      (let ((gp-2 (new 'stack-no-clear 'vector))
+            (s5-2 (new 'stack-no-clear 'vector))
+            )
+        (set! (-> gp-2 quad) (-> self root-override trans quad))
+        (set! (-> gp-2 w) 4096.0)
+        (when (sphere-in-view-frustum? (the-as sphere gp-2))
```

Comment:
interesting, I think this means the spider won't shoot if you can't see it.

Reply:
that wouldn't be too surprising -- that's a common thing across all the jak games.  It'll be interesting to see if this same pattern is used in the future

Reply:
I'm pretty sure looking away from a spider is a strat used in speedruns when shooting the gnawers.